src/main/webapp/components/services/utils.js

```diff
@@ -56,6 +56,58 @@
           return false;
         }
         return true;
+      },
+
+      setValue: function setValue(object, path, value) {
+        var a = path.split('.');
+        var o = object;
+        for (var i = 0; i < a.length - 1; i++) {
+          var n = a[i];
+          if (n in o) {
+            o = o[n];
+          } else {
+            o[n] = {};
+            o = o[n];
+          }
+        }
+        o[a[a.length - 1]] = value;
+      },
+
+      getValue: function getValue(object, path) {
+        var o = object;
+        path = path.replace(/\[(\w+)\]/g, '.$1');
+        path = path.replace(/^\./, '');
+        var a = path.split('.');
+        while (a.length) {
+          var n = a.shift();
+          if (n in o) {
+            o = o[n];
+          } else {
+            return;
+          }
+        }
+        return o;
+      },
+
+      makeResourceDescriptionMap: function makeResourceDescriptionMap(map){
```

Comment:
I am not sure if this function should go to utils. I usually prefer to keep only non-domain related functions in utils, but if you choose to keep those also then it is ok.